translate_test.go

```diff
@@ -1993,6 +2327,94 @@ func TestTranslate3_3to3_2(t *testing.T) {
 	assert.Error(t, err)
 }
 
+func TestTranslate3_4to3_3(t *testing.T) {
+	emptyConfig := types3_4.Config{
+		Ignition: types3_4.Ignition{
+			Version: "3.4.0",
+		},
+	}
+
+	_, err := v34tov33.Translate(emptyConfig)
+	if err != nil {
+		t.Fatalf("Failed translation: %v", err)
+	}
+
+	res, err := v34tov33.Translate(nonexhaustiveConfig3_4)
+	if err != nil {
+		t.Fatalf("Failed translation: %v", err)
+	}
+	assert.Equal(t, nonexhaustiveConfig3_3, res)
+
+	_, err = v34tov33.Translate(types3_4.Config{
+		Ignition: types3_4.Ignition{
+			Version: "3.4.0",
+		},
+		Storage: types3_4.Storage{
+			Luks: []types3_4.Luks{
+				{
+					Clevis: types3_4.Clevis{
+						Tang: []types3_4.Tang{
+							{
+								URL:           "http://example.com",
+								Thumbprint:    util.StrP("z"),
+								Advertisement: util.StrP("{payload: 'hello world'}"),
+							},
+						},
+					},
+				},
+			},
+		},
+	})
+	assert.Error(t, err)
+
+	_, err = v34tov33.Translate(types3_4.Config{
+		Ignition: types3_4.Ignition{
+			Version: "3.4.0",
+		},
+		Storage: types3_4.Storage{
+			Luks: []types3_4.Luks{
+				{
+					Discard: util.BoolP(true),
+				},
+			},
+		},
+	})
+	assert.Error(t, err)
+
+	_, err = v34tov33.Translate(types3_4.Config{
+		Ignition: types3_4.Ignition{
+			Version: "3.4.0",
+		},
+		Storage: types3_4.Storage{
+			Luks: []types3_4.Luks{
+				{
+					OpenOptions: []types3_4.OpenOption{"foo"},
+				},
+			},
+		},
+	})
+	assert.Error(t, err)
+
+	_, err = v34tov33.Translate(types3_4.Config{
+		Ignition: types3_4.Ignition{
+			Version: "3.4.0",
+		},
+		Storage: types3_4.Storage{
+			Files: []types3_4.File{
+				{
+					FileEmbedded1: types3_4.FileEmbedded1{
+						Contents: types3_4.Resource{
+							Source: util.StrP("arn:aws:s3:us-west-1:123456789012:accesspoint/test/object/some/path"),
+						},
+					},
+				},
+			},
+		},
+	})
+	assert.Error(t, err)
+
```

Comment:
Should also check that special mode bits are rejected.

also, nit: blank line :cry: